src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,196 @@
+
+    /** Display the popup on trigger focus */
+    focus: PropTypes.bool,
+
+    /** Header displayed above the content in bold */
+    header: PropTypes.string,
+
+    /** Display the popup on mouse over the trigger */
+    hover: PropTypes.bool,
+
+    /** Invert the colors of the popup */
+    inverted: PropTypes.bool,
+
+    /** Positionning for the popover */
+    position: PropTypes.string,
+
+    /** Popup size */
+    size: PropTypes.string,
+
+    /** Element to be rendered in-place where the popup is defined. */
+    trigger: PropTypes.node,
+
+    /** Popup width */
+    width: PropTypes.string,
+  }
+
+  static defaultProps = {
+    position: 'top left',
+  }
+
+  static _meta = _meta
+  static Content = PopupContent
+  static Header = PopupHeader
+
+  state = {}
+
+  getPosition(coords) {
+    if (!coords) return
+
+    const positions = this.props.position.split(' ')
+    const style = { position: 'absolute' }
+
+    if (includes(positions, 'right')) {
+      style.right = Math.round(document.documentElement.clientWidth - (coords.right + window.pageXOffset)) + 'px'
+      style.left = 'auto'
+    }
+
+    if (includes(positions, 'left')) {
+      style.left = Math.round(coords.left + window.pageXOffset) + 'px'
+      style.right = 'auto'
+    }
+
+    if (includes(positions, 'top')) {
+      style.bottom = Math.round(document.documentElement.clientHeight - (coords.top + window.pageYOffset)) + 'px'
+      style.top = 'auto'
+    }
+
+    if (includes(positions, 'bottom')) {
+      style.top = Math.round(coords.bottom + window.pageYOffset) + 'px'
+      style.bottom = 'auto'
+    }
+
+    return style
+  }
+
+  onOpen = (event) => {
+    this.setState({
+      coords: findDOMNode(this).getBoundingClientRect(),
+    })
+  }
+
+  componentWillMount() {
+    const portalProps = {}
+
+    if (this.props.click) {
+      portalProps.openOnTriggerClick = true
+      portalProps.closeOnDocumentClick = true
+    }
+    if (this.props.hover) {
+      portalProps.openOnTriggerMouseOver = true
+      portalProps.closeOnTriggerMouseLeave = true
+    }
+    if (this.props.focus) {
+      portalProps.openOnTriggerFocus = true
+      portalProps.closeOnTriggerBlur = true
+    }
+
+    portalProps.onOpen = this.onOpen
+
+    this.setState({ portalProps })
+  }
+
+  renderContent() {
+    const children = []
+    let index = 0
+    const { header, content } = this.props
+
+    if (header) {
+      children.push(<PopupHeader key={index++}>{header}</PopupHeader>)
+    }
+
+    if (content) {
+      if (isString(content)) {
+        children.push(<PopupContent key={index++}>{content}</PopupContent>)
+      } else {
+        children.push(content)
+      }
+    }
+
+    return children
+  }
+
+  render() {
+    const {
+      basic,
+      children,
+      className,
+      flowing,
+      fluid,
+      inverted,
+      position,
+      size,
+      trigger,
+      width,
+    } = this.props
+
+    const { coords } = this.state
```

Comment:
Let's add `portalProps` used below as well.  I should add some lint rules for this since we destructure props/state always.